Решение к 1 уроку

hmzxpDev opened the pull request.

hmzxpDev committed:
Решение к 1 уроку

hmzxpDev committed:
Решение ДЗ 2 урока

hmzxpDev committed:
ДЗ к 3 уроку

Не успел доделать пагинацию, не обращайте внимание на нижнюю часть

7rows commented on src/App.vue:
```diff
@@ -0,0 +1,130 @@
+      <addingCosts @formSend="pushEL" />
+      <paymentList :list="data" />
+      <paginate
+        :page-count="2"
```
а почему вы передаете только 2 ?

7rows replied:
у вас в компонент вы должны передать кол-во отображаемых элементов, а кол-во страниц должно само определяться, 
но так как это на звездочку . то это так, на подумать
вы должны per-page передавать еще в компонент

hmzxpDev committed:
сделал дз к 4 уроку

Вот и пригодился fecth))

hmzxpDev committed:
сделал дз к 5 уроку

1. реализовал заполнение формы по ссылке
2. Связал url и пагинацию
3. Попробовал transition, прощу прощения за такую кривую анимацию, суть была что бы попробовать.
4. Реализовал страницу NotFound

Касательно минусов своей работы :
1. Не смог использовать хуки роутинга, т.к. имею только одну страницу, особо не разгуляться
2.  По моему мнению - я зря сделал два роута в одну страницу(пагинация и add payment), но задача была сделать разные url, но страница была одна
3. Захламленный код. Я посмотрел как реализуют проекты другие разработчики, там в компонентах папки со отсортированными компонентами,  а у меня не правильная структура. Но я не буду на этом заморачиваться, потому что от урока к уроку - порой нужно кардинально переделывать страницы. По этому для удобства пытался максимально комментировать код

hmzxpDev committed:
Сделал дз к 6 уроку

До последнего не хотел усложнять логику приложения. Но все равно сделал доп плагин, подключил его, из него вызвал плагины vuex и modal. Надеюсь этого будет достаточно

7rows commented on src/plugins/contextMenu/index.js:
```diff
@@ -0,0 +1,31 @@
+                store.commit("deleteCost", value);
+            },
+            // отправляет данные в плагин модального окна, что бы открыл редактирование
+            openModal(value) {
```
Не совсем верный подход, а что , если у нас не будет модального окна ? Лучше проброс данных делать внутри методов компонентов, в которых вы слушаете его. 
Но идея может быть и решение рабочее, просто они не расширяемое и всегда нас заставляет думать не только о контекстном меню, но и о модальном окне

hmzxpDev replied:
Если не будет модального окна то будет другая структура взаимодействия. Но так сложилось что на уроке показали модальное окно, оно мне понравилось и я реализовал через него. Касательно передачи данных, я мог через стор, шину событий, пропсами, но я учусь и делаю всегда по разному что бы научиться.
По поводу расширяемости соглашусь, но я где то читал, что нерасширяемый код имеет место быть т.к. существует вероятность когда заказчик не приобретет поддержку или не захочет расширять приложения, тогда - мы сделаем лишнюю работу.

hmzxpDev committed:
Сделал дз к 7 уроку

Добрый день, протестил свой компонент по добавлению расходов. Не стал тратить время на тестирование других компонентов т.к. считаю что научился основам тестирования vue приложения, следующий курс через несколько дней и категорически нужно идти дальше. Не смог найти решения проблемы с catch (который я использовал для скрытия ошибки "NavigationDublicate") Если вы опишите решение буду вам очень благодарен. Так же решил не тратить время на тест роутинга и собственного плагина. P.S. в задании попросили сделать 5 тестов, изначально я сделал больше 7 тестов, а потом решил что можно не дублировать кучу кода и максимально сжал до 4 тестов.

hmzxpDev committed:
сделал дз к 8 уроку

Добрый день, единственная проблема это v-dialog внутри v-menu, внутри addingcosts компоненте. не закрывается меню во время открытия модального окна, если редактировать затрату
